src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/crud-table/crud-form/crud-form.component.ts

```diff
@@ -32,28 +37,46 @@ export class CrudFormComponent implements OnInit {
     private dataGatewayService: DataGatewayService,
     private activatedRoute: ActivatedRoute,
     private taskWrapper: TaskWrapperService,
-    private location: Location
+    private location: Location,
+    private router: Router
   ) {}
 
   ngOnInit(): void {
     this.activatedRoute.data.subscribe((data: any) => {
       this.resource = data.resource;
       this.dataGatewayService.list(`ui-${this.resource}`).subscribe((response: any) => {
-        this.title = response.forms[0].control_schema.title;
-        this.uiSchema = response.forms[0].ui_schema;
-        this.controlSchema = response.forms[0].control_schema;
+        for (let i = 0; i < response.forms.length; i++) {
+          if (this.router.url.includes(response.actions[i].routerLink)) {
+            this.urlFormName = this.router.url.substring(this.router.url.lastIndexOf('/') + 1);
+            this.methodType = response.forms[i].method_type;
+            this.title = response.forms[i].container_schema.control_schema.title;
+            this.uiSchema = response.forms[i].container_schema.ui_schema;
+            this.controlSchema = response.forms[i].container_schema.control_schema;
+          }
+        }
```

Comment:
Why don't we simply generate data structures in Python that are ready to be consumed by the UI layer without any further processing/mapping? I mean, no need to manually cherry-pick, map and transform fields. Just a common Python class - Typescript class, so this becomes like:

```ts
      this.dataGatewayService.list(`ui-${this.resource}`).subscribe((response: any) => this.whatever = response.whatever)
```

And if we are lazy (and IMHO good devels should be lazy yet lovers of their craft), there are even libraries to transform `Python typings <-> JSON Schema <-> Typescript Classes`.